ssh_tunnel.go

```diff
@@ -95,9 +94,15 @@ func (tunnel *SSHTunnel) forward(localConn net.Conn) {
 	var (
 		serverConn   *ssh.Client
 		err          error
-		attemptsLeft int = MAX_CONNECTION_ATTEMPTS
+		attemptsLeft int
 	)
 
+	if tunnel.MaxConnectionAttempts == 0 {
```

Comment:
You don't need this logic, instead just use:

```go
attemptsLeft = tunnel.MaxConnectionAttempts
```

Then below (since the loop always guarantees one iteration):

```go
if attemptsLeft <= 0 {
```

Reply:
Also, GitHub doesn't let me comment on line 112:

```go
tunnel.logf("server dial error: %+v, [%d]", err, attemptsLeft)
```

This isn't helpful as `attemptsLeft` will always be 0 (or less than 0). I think you mean:

```go
tunnel.logf("server dial error: %w: exceeded %d attempts", err, tunnel.MaxConnectionAttempts)
```

Reply:
Nice. Updated the code